### DISCUSSION TITLE
Any plans to upgade minio to decrease the minimum multipart upload chunk size?

### DISCUSSION THREAD

**Discussion by patchthecode** [minio/minio#13456] [Q&A] [answered]
I think the minimum chunk size for multipart upload is 5mb.
So I have 1 question

Is there any limitations that is preventing you from doing this? Or is that 5mb limitation strictly to be compatible with aws s3?

The use case is to build a social media app.  Taking Twitter as an example, they allow small chunk sizes for the following reason.

_________________________________
A 3 MB (fo example) file could be split into 3 chunks of size 1 MB, and uploaded using 3 APPEND command requests. After the entire file is uploaded, the next step is to call the FINALIZE command.

There are a number of advantages of uploading a media file in small chunks:

    1. ✅ Improved reliability and success rates under low bandwidth network conditions 
    2. Uploads can be paused and resumed
    3. File chunks can be retried individually
    4. ✅ Ability to tune chunk sizes to match changing network conditions e.g on cellular clients
_________________________________

I think that #1 & #4 makes a good use case.  Is it possible this update can be done while still being s3 compatible?
Or if I had to fork min.io to accomplish this, would this be a huge change for me to do?

**Comment by harshavardhana** [comment, marked as answer]
Minimum size is governed by S3 API. We won't be changing it until AWS S3 does it.

**Reply by patchthecode** [reply]
ok understood but

I think I will try to do a fork and modify the code, but currently I do not know how heavy the task is.

1. Is changing the size a massive change?
2. Is there any technical problems with going lower than 5mb?  or is the only reason to be compliant with AWS S3 API?

**Reply by klauspost** [reply]
1. Maybe just https://github.com/minio/minio/blob/974073a2e5203ea34976c76336cbd212c78fa6e3/cmd/utils.go#L172 - of course many tests will fail.
2. No idea. It is not something we tested. Main initial motivation was compatibility, but other things may depend on it.

**Reply by patchthecode** [reply]
Thanks for the starting point!  I really appreciate it
